src/controls/mod.rs

```diff
@@ -269,6 +285,11 @@ impl Program for ControlState {
                     self.action_queue.borrow_mut().push(action);
                 }
             }
+            Message::Gravity(message) => {
+                if let Some(action) = self.gravity_control.update(message) {
+                    self.action_queue.borrow_mut().push(action);
```

Comment:
You should extract a method `queue_action()` for this

Reply:
Indeed